### PR TITLE
Deeper nulltestcase

### PR DESCRIPTION
**Pull request by JosiahRegencia**
This is for the added feedback on issue #117 
I wasn't really sure what was meant. Made this as temporary solution instead.
What it does is that savant searches through the main tables in the database and looks for duplicates.
Build passed in: https://travis-ci.org/Doveps/mono/builds/349770733

### CODE REVIEW

**Comment by greenmoss** [comment]
Thanks, following up with Slack conversation:

> I would avoid db calls and creating special api endpoints just for the tests
The api should include only end points we will need as a normal savant user
At least for the “real” test
If you want to reset db between tests that is totally ok, and you can use direct db calls to do that

> Db calls in tests are ok if you need to reset
But avoid them for generating or retrieving test results
For those use only the api
Try to test the same way an end user would do it
For instance an end user might hit api /blah
Although /blah does db stuff a user doesn’t see that
User only sees api input and output
So tests should do the same

> If we see errors as a result of api calls we write a test to generate the same error using the same api call
Then fix it and the test now passes
Yeah that can be challenging
It will make the api stronger

Feedback for the PR 😄

File: savant/app/query.py [hunk 1]
```diff
@@ -153,3 +153,22 @@ def new_users(self):
         self.conn.commit()
         
         return self.new_users
+
```
Comment (greenmoss) [inline savant/app/query.py:156]: Following up on comments on #141, assuming this branch is part of it. Strange, I don't see those changes here. I am confused 😄 

Anyway, following those thoughts, we should not need `null_cases` here.

File: savant/app/results.py [hunk 2]
```diff
@@ -11,9 +11,9 @@
 def create_flavors():
     filenames = request.files.getlist('files[]')
     get_scanned.get_items(filenames)
-    # query.record_flavors()
     que_flavors = query.Query()
     que_flavors.record_flavors()
+    que_flavors.null_cases()
```
Comment (greenmoss) [inline savant/app/results.py:16]: Same, I don't think we need this.

File: savant/app/results.py [hunk 3]
```diff
@@ -35,6 +35,8 @@ def compare():
     with io.open(path + "/mono/savant/app/" + json_file, 'w', encoding='utf-8') as data:
         data.write(unicode(new_packages))
 
+    que_compare.null_cases()
```
Comment (greenmoss) [inline savant/app/results.py:38]: Don't need this either

File: savant/tests/test_basic.py [hunk 4]
```diff
@@ -32,19 +34,10 @@ def test_compare(self):
...
-        cur.execute("select store_debs(%s, %s ,%s, %s)", ('stat', None, 'test', 'test'))
-        conn.commit()
-        cur.execute("select store_debs(%s, %s ,%s, %s)", ('stat', None, 'test', 'test'))
-        conn.commit()
-        cur.execute("select count(*) from debs where stat=%s and name is null and version=%s and architecture=%s", ('stat', 'test', 'test'))
-        count = cur.fetchone()
-        self.assertEqual(count[0], 1)
-
-
+    #def test_nullcases(self):
```
Comment (greenmoss) [inline savant/tests/test_basic.py:37]: I feel like I'm doing github wrong. I saw this set of changes on an earlier commit, but this was already fixed. Very strange. Is this PR still active, or should we close it?

**Review by greenmoss — CHANGES_REQUESTED** [review]

File: savant/tests/test_basic.py (hunk 4 above)
Comment (JosiahRegencia) [reply savant/tests/test_basic.py:37]: I think it's my fault. Probably shouldn't have created a PR yet.

Comment (JosiahRegencia) [reply savant/tests/test_basic.py:37]: @greenmoss I don't think this PR is active. can we delete this?

Comment (JosiahRegencia) [reply savant/tests/test_basic.py:37]: Just realized it now

**Pull request closed without merging**